### PR TITLE
ci: 👷 commit-lint.yml

### LINKED ISSUE #128: [ci] :construction_worker:  Conventional Commits check with every PR
It would be nice to have this CI feature for automatic check and warn to user in PR section. I can added relevant CI code quickly.

Source : https://www.conventionalcommits.org/en/v1.0.0/

### PR DESCRIPTION
**Pull request by onuralpszr**
Conventional Commits fix : https://github.com/t0xic0der/nvidia-auto-installer-for-fedora-linux/issues/128

### CODE REVIEW

**Comment by onuralpszr** [comment]
@t0xic0der I fixed the commit for msg for pass the test now you can see it works as well. We can merge it.

File: .github/workflows/commit-lint.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Conventional Commits
+on: [pull_request]
+
+jobs:
+  lint-commits:
+    name: Lint Commits
```
Comment (gridhead) [inline .github/workflows/commit-lint.yml]:
```suggestion
    name: Lint commits
```

File: .github/workflows/commit-lint.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2.3.1
+        with:
+          fetch-depth: 0
+
+      - name: Install Commit Linting Tool 🔧
```
Comment (gridhead) [inline .github/workflows/commit-lint.yml]:
```suggestion
      - name: Install commit linting tool 🔧
```

File: .github/workflows/commit-lint.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        run: npm install --save-dev @commitlint/{cli,config-conventional}
+
+      - name: Set Linting Config to Conventional Commits spec 🔧
```
Comment (gridhead) [inline .github/workflows/commit-lint.yml]:
```suggestion
      - name: Set linting config to Conventional Commits spec 🔧
```

**Review by gridhead — CHANGES_REQUESTED** [review]
Some nitpicks really but yeah, please do these :)

**Comment by onuralpszr** [comment]
@t0xic0der  all done and LGTM and all passed.

**Review by gridhead — APPROVED** [review]
LGTM

**Pull request merged**